onnxruntime/core/providers/cpu/tensor/scatter_nd.cc

```diff
@@ -0,0 +1,156 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+#include "scatter_nd.h"
+
+namespace onnxruntime {
+
+ONNX_CPU_OPERATOR_KERNEL(
+    ScatterND,
+    11,
+    KernelDefBuilder()
+        .TypeConstraint("T",    DataTypeImpl::AllTensorTypes())
+        .TypeConstraint("Tind", DataTypeImpl::GetTensorType<int64_t>()),
+    ScatterND);
+
+template<typename Tind>
+Status ScatterNDBase::PrepareForCompute(OpKernelContext* context, Prepare& p) const {
+
+  auto input_tensor  = context->Input<Tensor>(0);
+  auto indice_tensor = context->Input<Tensor>(1);
+  auto update_tensor = context->Input<Tensor>(2);
+  ORT_ENFORCE(input_tensor  != nullptr);
+  ORT_ENFORCE(indice_tensor != nullptr);
+  ORT_ENFORCE(update_tensor != nullptr);
+
+  auto input_shape   = input_tensor->Shape();
+  auto indice_shape  = indice_tensor->Shape();
+  auto update_shape  = update_tensor->Shape();
+  if (indice_shape.NumDimensions() == 0) {
+    return ORT_MAKE_STATUS(ONNXRUNTIME, INVALID_ARGUMENT,
+      "indices tensor must has rank larger than 0");
+  }
```

Comment:
ScatterND takes three **inputs data tensor of rank r >= 1**, indices tensor of rank q >= 1, and **updates tensor of rank q + r - indices.shape[-1] - 1**.  Shall we check input_shape and update_shape as well?

Reply:
Thanks for the catch. Added checks for input_shape. It is possible for update rank to be zero.